library/src-bootstrapped/scala/quoted/Expr.scala

```diff
@@ -18,7 +18,8 @@ abstract class Expr[+T] private[scala] {
    *  ```
    */
   final def matches(that: Expr[Any])(using qctx: QuoteContext): Boolean =
-    !scala.internal.quoted.Expr.unapply[EmptyTuple, EmptyTuple](this)(using that, qctx).isEmpty
+    val ExprMatch = qctx.asInstanceOf[scala.internal.quoted.QuoteContextInternal].ExprMatch
+    ExprMatch.unapply[EmptyTuple, EmptyTuple](this)(using that).nonEmpty
```

Comment:
From the usage, it seems to me `ExprMatch` and `TypeMatch` does not make it simpler. Do we have places where they are used as extractors?

Reply:
This is the only place in the library where it is needed. But I plan to move this implementation out of the library and then I might not even need to have this call and could call the tree pattern matcher directly.

This is another method that should only be called from code generated by the compiler.